substrate/primitives/trie/src/recorder.rs

```diff
@@ -231,15 +228,43 @@ impl<H: Hasher> Recorder<H> {
 	}
 }
 
+impl<H: Hasher> crate::ProofSizeProvider for Recorder<H> {
+	fn estimate_encoded_size(&self) -> usize {
+		Recorder::estimate_encoded_size(self)
+	}
+}
+
+impl<H: Hasher> crate::ProofSizeProvider for &Recorder<H> {
+	fn estimate_encoded_size(&self) -> usize {
+		Recorder::estimate_encoded_size(*self)
+	}
+}
+
 /// The [`TrieRecorder`](trie_db::TrieRecorder) implementation.
-struct TrieRecorder<H: Hasher, I> {
-	inner: I,
+pub struct TrieRecorder<'a, H: Hasher> {
+	inner: MutexGuard<'a, RecorderInner<H::Out>>,
```

Comment:
This is very similar to before, just that the `I` generic parameter is removed. I do not think its needed, since the `TrieRecorder` implementation was bound to the [`RecordInner` type anyway](https://github.com/skunert/polkadot-sdk/blob/e6f5e23b0f894954c0f04ba379966f2466f51de6/substrate/primitives/trie/src/recorder.rs#L286).